specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2023-05-01-privatepreview/AlertProcessingRules.json

```diff
@@ -0,0 +1,865 @@
+          "$ref": "#/definitions/Conditions"
+        },
+        "schedule": {
+          "description": "Scheduling for alert processing rule.",
+          "$ref": "#/definitions/Schedule"
+        },
+        "actions": {
+          "type": "array",
+          "description": "Actions to be applied.",
+          "items": {
+            "$ref": "#/definitions/Action"
+          }
+        },
+        "description": {
+          "type": "string",
+          "description": "Description of alert processing rule."
+        },
+        "enabled": {
+          "type": "boolean",
+          "default": true,
+          "description": "Indicates if the given alert processing rule is enabled or disabled."
+        }
+      }
+    },
+    "Action": {
+      "type": "object",
+      "description": "Action to be applied.",
+      "discriminator": "actionType",
+      "required": [
+        "actionType"
+      ],
+      "properties": {
+        "actionType": {
+          "type": "string",
+          "description": "Action that should be applied.",
+          "enum": [
+            "AddActionGroups",
+            "RemoveAllActionGroups",
+            "CorrelateAlerts"
+          ],
+          "x-ms-enum": {
+            "name": "actionType",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "AddActionGroups": {
+      "x-ms-discriminator-value": "AddActionGroups",
+      "type": "object",
+      "description": "Add action groups to alert processing rule.",
+      "required": [
+        "actionGroupIds"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/Action"
+        }
+      ],
+      "properties": {
+        "actionGroupIds": {
+          "description": "List of action group Ids to add to alert processing rule.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "RemoveAllActionGroups": {
+      "x-ms-discriminator-value": "RemoveAllActionGroups",
+      "type": "object",
+      "description": "Indicates if all action groups should be removed.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Action"
+        }
+      ]
+    },
+    "CorrelateAlerts": {
+      "x-ms-discriminator-value": "CorrelateAlerts",
+      "type": "object",
+      "description": "Add logic for alerts correlation.",
+      "correlateBy": {
+        "type": "array",
+        "items": {
+          "type": "#/definitions/CorrelatedBy"
+        },
+        "description": "The list of conditions for the alerts correlations."
+      },
+      "correlationInterval": {
+        "type": "string",
+        "format": "duration",
+        "description": "The required duration (in ISO8601 format) for the alerts correlation."
+      },
+      "priority": {
+        "type": "integer",
+        "format": "int32",
+        "description": "The priority of this correlation."
+      },
+      "suppressNotificationForChildAlerts": {
```

Comment:
>"suppressNotificationForChildAlerts": { [](http://example.com/codeflow?start=6&length=39)

Can this be x-ms-enum? Something like:

childAlertsNotificateState with values ["Suppressed", "Active"] etc. That allows you to add more states if needed in future without breaking things.